ue5osc/__init__.py

```diff
@@ -1,21 +1,103 @@
 from pythonosc import udp_client
+from time import sleep
+
+from ue5osc.osc_dispatcher import DispatchHandler
+from pythonosc.osc_server import BlockingOSCUDPServer
+from PIL import Image
+
+import matplotlib.pyplot as plt
 
 
 class OSCSender:
-    def __init__(self, ip: str, port: int):
+    """Handles interaction between the UE5 environment and the a program."""
+
+    # Initializing the OSC communicator
+    def __init__(self, ip: str, port: int, directory: str):
+        self.path = directory
+        self.location_handler = DispatchHandler()
         self.client = udp_client.SimpleUDPClient(ip, port)
+        self.server = BlockingOSCUDPServer((ip, 7001), self.location_handler.dispatcher)
+        self.imgnumber = 0
+
+    # Starts the OSC server
+    def start_server(self):
+        self.server.serve_forever()
+
+    def get_project_name(self):
+        """Returns the name of the current connected project."""
+        self.client.send_message("/get/project", 0.0)
+        self.location_handler.delay("Project")
+        name = self.location_handler.project_name
+        self.location_handler.project_name = None

```

Comment:
This should probably be a single function call. Let the `wait`/`delay` method handle the details.